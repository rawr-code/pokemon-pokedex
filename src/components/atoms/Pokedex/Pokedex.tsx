// Utils
import { createTestIDs, getColor, cn } from '@utils'
import { FC } from 'react'

const ownTestIDs = createTestIDs('Pokedex', ['root'])

// Types
interface PokedexProps {
  children: React.ReactNode
  className?: string
  classes?: {
    root?: string
  }
  testIDs?: { [k in keyof typeof ownTestIDs]?: string | undefined }
}

const Light = ({ className }: { className: string }) => (
  <div
    className={cn(
      'relative z-10 mr-2 mt-2 h-4 w-4 rounded-full border-2 border-pokedex-border before:absolute before:bottom-0 before:left-1/2 before:h-[90%] before:w-[90%] before:-translate-x-1/2 before:rounded-full before:bg-black/20 before:content-[""] after:absolute after:left-[10%] after:top-[10%] after:h-[60%] after:w-[60%] after:rounded-full',
      className,
    )}
  />
)

const Pokedex: FC<PokedexProps> = ({
  children,
  className = '',
  classes = {},
  testIDs = {},
}) => {
  return (
    <main
      className={cn(
        'flex h-full w-full items-center justify-center',
        className,
        classes.root,
      )}
      data-testid={testIDs.root || ownTestIDs.root}
    >
      <div className="relative h-full w-[540px] overflow-hidden border-pokedex-border bg-pokedex-body pt-8 sm:h-[720px] sm:rounded-[20px] sm:border-2 sm:p-8">
        {/* BlueCircle */}
        <div className='absolute left-7 top-4 z-[60] mt-0.5 flex h-16 w-16 items-center justify-center rounded-full border-2 border-pokedex-border bg-white after:h-12 after:w-12 after:rounded-full after:border-2 after:border-pokedex-border after:bg-cyan-400 after:content-[""] sm:left-10' />
        {/* SpeakerWrapper */}
        <div className="absolute left-1/2 top-3 z-[100] flex -translate-x-1/2">
          <div className="mx-1 h-2 w-12 rounded-md border-2 border-pokedex-border bg-pokedex-wrapper" />
          <div className="mx-1 h-2 w-12 rounded-md border-2 border-pokedex-border bg-pokedex-wrapper" />
        </div>
        {/* LightWrapper */}
        <div className="absolute right-4 top-0 flex sm:right-8">
          <Light className="bg-[#ee0707] after:bg-[#ee0707]" />
          <Light className="bg-[#dde214] after:bg-[#dde214]" />
          <Light className="bg-[#91f913] after:bg-[#91f913]" />
        </div>
        {/* Border */}
        <div className="relative h-full w-full rounded-b-2xl border-pokedex-border bg-pokedex-wrapper pt-5 sm:border-2 sm:p-5">
          <div className="absolute left-0 top-5 z-50 before:absolute before:left-0 before:top-0 before:h-16 before:w-24 before:bg-pokedex-wrapper after:absolute after:-top-6 after:left-16 after:h-16 after:w-24 after:-rotate-[41deg] after:bg-pokedex-wrapper sm:left-2" />
          <div className="absolute -left-1.5 top-0 z-50 before:absolute before:-top-4 before:left-0 before:h-20 before:w-24 before:border-b-2 before:border-pokedex-border before:bg-pokedex-body after:absolute after:-top-6 after:left-16 after:h-16 after:w-24 after:-rotate-[41deg] after:border-b-2 after:border-pokedex-border after:bg-pokedex-body sm:-left-0.5" />
          <div className="h-full w-full overflow-hidden rounded-b-md bg-white">
            {children}
          </div>
        </div>
      </div>
    </main>
  )
}

export default Pokedex
export { ownTestIDs as testIDs }
