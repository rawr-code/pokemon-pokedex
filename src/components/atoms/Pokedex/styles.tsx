import tw, { css, styled } from 'twin.macro'

export const Wrapper = tw.div`relative w-[540px] h-[720px] bg-pokedex-body rounded-[20px] p-8 overflow-hidden border-2 border-pokedex-border`
export const Border = tw.div`relative w-full h-full border-2 border-pokedex-border bg-pokedex-wrapper rounded-b-2xl p-5`
export const Screen = tw.div`w-full h-full rounded-b-md bg-white`

export const BlueCircle = tw.div`
    absolute z-10 w-16 h-16 bg-white border-2 border-pokedex-border rounded-full top-4 mt-0.5 left-10 flex items-center justify-center
    after:content-[""] after:w-12 after:h-12 after:bg-cyan-400 after:border-2 after:border-pokedex-border after:rounded-full
`
export const LightWrapper = tw.div`absolute right-8 top-0 flex`

interface StyledLightProps {
  color: string
}

export const StyledLight = styled.div<StyledLightProps>`
  ${({ color }) => css`
    background-color: ${color};
    &::after {
      background-color: ${color};
    }
  `}
  ${tw`
    relative z-10 h-4 w-4 mt-2 mr-2 border-2 border-pokedex-border rounded-full
    before:content-[""] before:absolute before:w-[90%] before:h-[90%] before:bottom-0 before:left-1/2 before:-translate-x-1/2 before:rounded-full before:bg-black/20
    after:content-[""] after:absolute after:w-[60%] after:h-[60%] after:top-[10%] after:left-[10%] after:rounded-full
    `}
`

export const Light = ({ color }: StyledLightProps) => (
  <StyledLight color={color} />
)

export const SpeakerWrapper = tw.div`absolute top-3 left-1/2 -translate-x-1/2 flex`
export const Speaker = tw.div`bg-pokedex-wrapper w-12 h-2 border-2 border-pokedex-border rounded-md mx-1`

export const CornerBorder = tw.div`
  absolute left-2 top-5
  before:absolute before:left-0 before:top-0 before:bg-pokedex-wrapper before:w-24 before:h-16
  after:absolute after:-rotate-[41deg] after:left-16 after:-top-6 after:bg-pokedex-wrapper after:w-24 after:h-16
`

export const CornerBody = tw.div`
  absolute -left-0.5 top-0
  before:absolute before:left-0 before:-top-4 before:bg-pokedex-body before:w-24 before:h-20 before:border-b-2 before:border-pokedex-border
  after:absolute after:-rotate-[41deg] after:left-16 after:-top-6 after:bg-pokedex-body after:w-24 after:h-16 after:border-b-2 after:border-pokedex-border
`