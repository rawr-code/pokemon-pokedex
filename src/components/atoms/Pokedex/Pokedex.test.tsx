import { runCommonTests } from '@utils'

// Component
import Pokedex, { testIDs } from './Pokedex'

describe('Atom | Pokedex', () => {
  runCommonTests(
    Pokedex,
    { rootTestID: testIDs.root },
    {
      children: <></>,
    },
  )
})
