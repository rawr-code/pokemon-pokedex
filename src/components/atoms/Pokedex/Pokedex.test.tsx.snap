// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom | Pokedex should match snapshot 1`] = `
<div>
  <main
    class="flex h-full w-full items-center justify-center className classes.root"
    data-testid="Pokedex-root"
  >
    <div
      class="relative h-full w-[540px] overflow-hidden border-pokedex-border bg-pokedex-body pt-8 sm:h-[720px] sm:rounded-[20px] sm:border-2 sm:p-8"
    >
      <div
        class="absolute left-7 top-4 z-[60] mt-0.5 flex h-16 w-16 items-center justify-center rounded-full border-2 border-pokedex-border bg-white after:h-12 after:w-12 after:rounded-full after:border-2 after:border-pokedex-border after:bg-cyan-400 after:content-[""] sm:left-10"
      />
      <div
        class="absolute left-1/2 top-3 z-[100] flex -translate-x-1/2"
      >
        <div
          class="mx-1 h-2 w-12 rounded-md border-2 border-pokedex-border bg-pokedex-wrapper"
        />
        <div
          class="mx-1 h-2 w-12 rounded-md border-2 border-pokedex-border bg-pokedex-wrapper"
        />
      </div>
      <div
        class="absolute right-4 top-0 flex sm:right-8"
      >
        <div
          class="relative z-10 mr-2 mt-2 h-4 w-4 rounded-full border-2 border-pokedex-border before:absolute before:bottom-0 before:left-1/2 before:h-[90%] before:w-[90%] before:-translate-x-1/2 before:rounded-full before:bg-black/20 before:content-[""] after:absolute after:left-[10%] after:top-[10%] after:h-[60%] after:w-[60%] after:rounded-full bg-[#ee0707] after:bg-[#ee0707]"
        />
        <div
          class="relative z-10 mr-2 mt-2 h-4 w-4 rounded-full border-2 border-pokedex-border before:absolute before:bottom-0 before:left-1/2 before:h-[90%] before:w-[90%] before:-translate-x-1/2 before:rounded-full before:bg-black/20 before:content-[""] after:absolute after:left-[10%] after:top-[10%] after:h-[60%] after:w-[60%] after:rounded-full bg-[#dde214] after:bg-[#dde214]"
        />
        <div
          class="relative z-10 mr-2 mt-2 h-4 w-4 rounded-full border-2 border-pokedex-border before:absolute before:bottom-0 before:left-1/2 before:h-[90%] before:w-[90%] before:-translate-x-1/2 before:rounded-full before:bg-black/20 before:content-[""] after:absolute after:left-[10%] after:top-[10%] after:h-[60%] after:w-[60%] after:rounded-full bg-[#91f913] after:bg-[#91f913]"
        />
      </div>
      <div
        class="relative h-full w-full rounded-b-2xl border-pokedex-border bg-pokedex-wrapper pt-5 sm:border-2 sm:p-5"
      >
        <div
          class="absolute left-0 top-5 z-50 before:absolute before:left-0 before:top-0 before:h-16 before:w-24 before:bg-pokedex-wrapper after:absolute after:-top-6 after:left-16 after:h-16 after:w-24 after:-rotate-[41deg] after:bg-pokedex-wrapper sm:left-2"
        />
        <div
          class="absolute -left-1.5 top-0 z-50 before:absolute before:-top-4 before:left-0 before:h-20 before:w-24 before:border-b-2 before:border-pokedex-border before:bg-pokedex-body after:absolute after:-top-6 after:left-16 after:h-16 after:w-24 after:-rotate-[41deg] after:border-b-2 after:border-pokedex-border after:bg-pokedex-body sm:-left-0.5"
        />
        <div
          class="h-full w-full overflow-hidden rounded-b-md bg-white"
        />
      </div>
    </div>
  </main>
</div>
`;
